Zip Search Populating School Dropdown List

hannahbergam opened the pull request:
This PR utilizes the existing zip search to pull the schools by zip code into the dropdown with the other three 'default' options: Select a School, I don't teach at a school, and enter manually.

I elected to reuse the existing search but was able to pare down the search *call significantly because we are removing location search functionality. This way we are able to minimize the risk (and time) involved with writing a new search function, while streamlining performance. 

This exists currently in debouncedSearch from [SchoolAutocompleteDropdown here.](https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/templates/SchoolAutocompleteDropdown.jsx#L49)

The back end search is fairly involved and complex, and I am assuming most of these decisions were made for a clear (though perhaps not quite well documented) reason, and have elected to advise a full rewrite as a future performance enhancement rather than a P0 requirement here. [Here is the Jira ticket](https://codedotorg.atlassian.net/browse/ACQ-1717) for that followup work. 

I also added a sort functionality, an improvement over the existing search. This unfortunately highlights our duplicate schools issue, which should have a resolution soon from our RED team.

<img width="487" alt="Screenshot 2024-04-03 at 1 47 58 PM" src="https://github.com/code-dot-org/code-dot-org/assets/37230822/2c4028b0-9e93-4b54-a748-3a8c3efc62a5">


## Links

- spec: []()
- jira ticket: https://codedotorg.atlassian.net/browse/ACQ-1611

## Testing story

<!--
  Does your change include appropriate tests?
  If so, please describe how the tests included in this PR are sufficient.
  If not, please explain why this change does not need to be tested.
-->

<!-- Other aspects to consider. Delete any sections that are not relevant to your change. -->

## Deployment strategy

## Follow-up work

<!--
  List (ideally with Jira links) any clean-up or technical debt that will be addressed in future work.
-->

## Privacy

<!--
  1.	Does this change involve the collection, use, or sharing of new Personal Data?
  2.	Does this change involve a new or changed use or sharing of existing Personal Data?
-->

## Security

<!-- Link to Jira task(s) where sensitive security issues are discussed privately. -->

## Caching

## PR Checklist:

<!--
  The final step! Before you create your PR, double-check that everything is in order.
  Change [ ] to [X] during creation to check boxes.
-->

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

TurnerRiley reviewed (approved):
Looks great! Out of curiosity, do you think it would be worth making a ticket to add a Storybook or test file for SchoolZipSearch just to have in general to check that its returning the right value and that it handles having a combo of hardset values (e.g. "clickToAdd") and queried school values well?

bethanyaconnor commented on apps/src/templates/SchoolZipSearch.jsx:
```diff
@@ -53,6 +69,7 @@ export default function SchoolZipSearch({fieldNames}) {
             styleAsText={true}
             onClick={() => {
               setInputManually(false);
+              setSelectedSchoolNcesId('selectASchool');
```
nit: could this be moved into its own constant, used both here and in `SEARCH_DEFAULTS`?

bethanyaconnor reviewed (commented):
Are there any tests to add here?

hannahbergam commented:
> Are there any tests to add here?

@bethanyaconnor and @TurnerRiley I added a sort functionality, updated the description, and added a test with a fetch stub that verifies the sort is working! (Also there's a new screenshot to show that off)

bethanyaconnor commented on apps/test/unit/templates/SchoolZipSearchTest.jsx:
```diff
@@ -0,0 +1,79 @@
+};
+
+describe('SchoolDataInputs', () => {
+  allowConsoleWarnings();
```
curious why this is needed?
